recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -88,7 +88,8 @@ def requirements(self):
             self.requires("ms-gsl/4.0.0")
 
         if self.options.with_otlp_grpc:
-            self.requires("grpc/1.67.1", transitive_headers=True, transitive_libs=True)
+            # Version range matches arrow, which uses this as a dependency
+            self.requires("grpc/[>=1.50.0 <2]", transitive_headers=True, transitive_libs=True)
```

Comment:
Let's keep the lower limit as it was to avoid any unexpected behaviour:

```suggestion
            # Version range matches arrow, which uses this as a dependency
            self.requires("grpc/[>=1.67.1 <2]", transitive_headers=True, transitive_libs=True)
```